cycode/cli/files_collector/sca/base_restore_dependencies.py

```diff
@@ -13,19 +13,26 @@ def build_dep_tree_path(path: str, generated_file_name: str) -> str:
     return join_paths(get_file_dir(path), generated_file_name)
 
 
-def execute_command(
-    command: List[str],
+def execute_commands(
+    commands: List[List[str]],
     file_name: str,
     command_timeout: int,
     dependencies_file_name: Optional[str] = None,
     working_directory: Optional[str] = None,
 ) -> Optional[str]:
     try:
-        dependencies = shell(command=command, timeout=command_timeout, working_directory=working_directory)
-        # Write stdout output to the file if output_file_path is provided
+        all_dependencies = []
+
+        # Run all commands and collect outputs
+        for command in commands:
+            dependencies = shell(command=command, timeout=command_timeout, working_directory=working_directory)
+            all_dependencies.append(dependencies)  # Collect each command's output
+
+        # Write all collected outputs to the file if dependencies_file_name is provided
         if dependencies_file_name:
-            with open(dependencies_file_name, 'w') as output_file:
-                output_file.write(dependencies)
+            with open(dependencies_file_name, 'w') as output_file:  # Open once in 'w' mode to start fresh
+                for dependencies in all_dependencies:
+                    output_file.writelines(dependencies)
```

Comment:
Am I wrong about it?

```suggestion
                output_file.writelines(all_dependencies)
```

Reply:
After I checked again we do need a new line \n for the multi command generation, I move it upward inside the append code, then wI changed the actual file writing to `output_file.writelines(all_dependencies)` without for loop